vm/vm.go

```diff
@@ -2,12 +2,60 @@
 package vm
 
 import (
+	"sync"
+
+	peer "github.com/libp2p/go-libp2p-peer"
 	"github.com/libp2p/go-smart-record/ir"
 )
 
+// RecordValue determines the structure of data stored in a record.
+// Each peer has a private dataspace to store dicts.
+type RecordValue map[peer.ID]*ir.Dict
+
+type Record struct {
+	Key   string      // Key of the record
+	Value RecordValue // Root stores peers dictionaries in that key.
+}
+
 // Machine captures the public interface of a smart record virtual machine.
 type Machine interface {
-	Update(ir.Dict) error
-	Query(ir.Dict) error
-	Get() ir.Dict
+	Update(writer peer.ID, k string, update ir.Dict) error // Updates the dictionary in the writer's private space.
+	Get(k string) Record                                   // Get the full Record in a key
```

Comment:
This can just return RecordValue. This way you don't need to define the type Record. Simplification.

Reply:
BY THE WAY, don't worry about changing this. You can leave Record as is for now. Just a style comment for the future.

Reply:
The point being: The returned values should generally not repeat the input. The caller already knows the input.

Reply:
No, you are right. Too much redundancy. I rather change it now. Thanks!